view/next-project/src/components/purchaseorders/PurchaseItemNumModal.tsx

```diff
@@ -59,6 +61,22 @@ export default function PurchaseItemNumModal(props: PurchaseItemNumModalProps) {
     return initFormDataList;
   });
 
+  const [expenseByPeriods, setExpenseByPeriods] = useState<Expense[]>(props.expenseByPeriods);
+  console.log(expenseByPeriods);
+
+  const date = new Date();
+  const [selectedYear, setSelectedYear] = useState<number>(date.getFullYear());
+  const yearPeriods: YearPeriod[] = props.yearPeriods;
+
+  useEffect(() => {
+    const getExpenseByPeriodsUrl = process.env.CSR_API_URI + '/expenses/fiscalyear/' + selectedYear;
+    const getExpenseByPeriods = async (url: string) => {
+      const expenseByPeriodsRes: Expense[] = await get(url);
+      setExpenseByPeriods(expenseByPeriodsRes);
+    };
+    getExpenseByPeriods(getExpenseByPeriodsUrl);
```

Comment:
ここでformDataのexpenseIDの初期値を更新してください。(年度を変更した時のidのままのため)
```suggestion
    const getExpenseByPeriodsUrl = process.env.CSR_API_URI + '/expenses/fiscalyear/' + selectedYear;
    const getExpenseByPeriods = async (url: string) => {
      const expenseByPeriodsRes: Expense[] = await get(url);
      setExpenseByPeriods(expenseByPeriodsRes);
      expenseByPeriodsRes &&
        setFormData({ ...formData, expenseID: expenseByPeriodsRes[0].id || 1 });
    };
    getExpenseByPeriods(getExpenseByPeriodsUrl);
```